pkg/k8shandler/logstore.go

```diff
@@ -199,34 +214,36 @@ func newElasticsearchCR(cluster *logging.ClusterLogging, elasticsearchName strin
 		}
 	}
 
-	if logStoreSpec.NodeCount > 3 {
+	esNode := elasticsearch.ElasticsearchNode{
+		Roles:     []elasticsearch.ElasticsearchNodeRole{"client", "data", "master"},
+		NodeCount: logStoreSpec.NodeCount,
+		Storage:   logStoreSpec.ElasticsearchSpec.Storage,
+	}
+
+	// build Nodes
+	esNodes = append(esNodes, esNode)
 
-		masterNode := elasticsearch.ElasticsearchNode{
-			Roles:     []elasticsearch.ElasticsearchNodeRole{"client", "data", "master"},
-			NodeCount: 3,
-			Storage:   logStoreSpec.ElasticsearchSpec.Storage,
+	// if we had more than 1 es node before, we also want to enter this condition
+	if logStoreSpec.NodeCount > maximumElasticsearchMasterCount || len(existing.Spec.Nodes) > 1 {
+
+		// we need to check this because if we scaled down we can enter this block
+		if logStoreSpec.NodeCount > maximumElasticsearchMasterCount {
+			esNodes[0].NodeCount = maximumElasticsearchMasterCount
 		}
```

Comment:
This is a bit confusing to me. If we scaled down then how can logStoreSpec.NodeCount be greater than maximumElasticsearchMasterCount?

Reply:
I guess if someone had more master nodes before we introduced the new max?

Reply:
this cover the case where we scale up from 3, so where only the first part of that condition is true